@here/olp-sdk-dataservice-read/lib/layer/CatalogLayer.ts

```diff
@@ -0,0 +1,120 @@
+/*
+* Copyright (C) 2019 HERE Europe B.V.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+* License-Filename: LICENSE
+*/
+
+import {
+    ConfigApi,
+    CoverageApi,
+    MetadataApi,
+    QueryApi
+} from "@here/olp-sdk-dataservice-api";
+import {
+    ArtifactClient,
+    DataRequest,
+    HRN,
+    OlpClientSettings,
+    PartitionsRequest,
+    QuadKeyPartitionsRequest,
+    SchemaDetailsRequest,
+    SchemaRequest,
+    StatisticsClient,
+    StatisticsRequest,
+    SummaryRequest,
+    VersionedLayerClient,
+    VolatileLayerClient
+} from "..";
+
+export class CatalogLayer {
+    constructor(
+        private readonly model: ConfigApi.Layer,
+        private readonly settings: OlpClientSettings,
+        private readonly hrn: HRN,
+        private readonly layerId: string
+    ) {}
+
+    /**
+     * @returns this layer configuration
```

Comment:
Which "this"?
Please also add an explanation of the method.

Reply:
Fixed